### PR TITLE
Fix bug answer validate, delete category, delete word

### PR DESCRIPTION
**Pull request by longnv5994**

### CODE REVIEW

File: config/settings.yml [hunk 1]
```diff
@@ -4,6 +4,7 @@ per_page: 5
 build:
   repeat: 2
 min_words: 1
-ans_lesson: 4
+ans_lesson: 5
 word_filter: ["learned", "unlearned", "showall"]
-
+answer_default: 2
+max_answer: 4
```
Comment (anhkhang2304) [inline config/settings.yml]: OK

File: app/models/word.rb [hunk 2]
```diff
@@ -16,12 +16,25 @@ class Word < ActiveRecord::Base
...
   has_many :word_answers, dependent: :destroy
 
   accepts_nested_attributes_for :word_answers,
     reject_if: lambda {|a| a[:content].blank?}, allow_destroy: true
 
   validates :content, presence: true, length: {maximum: 140}
-end
+  validate :validate_answer
 
+  def validate_answer
```
Comment (nhunght) [inline app/models/word.rb]: private

File: app/models/word.rb [hunk 3]
```diff
@@ -16,12 +16,32 @@ class Word < ActiveRecord::Base
...
 
   accepts_nested_attributes_for :word_answers,
     reject_if: lambda {|a| a[:content].blank?}, allow_destroy: true
 
   validates :content, presence: true, length: {maximum: 140}
-end
+  validate :validate_answer
+
+  private
+    def validate_answer
```
Comment (dieunb) [inline app/models/word.rb]: indent

File: app/models/word.rb [hunk 4]
```diff
@@ -16,12 +16,32 @@ class Word < ActiveRecord::Base
...
+        errors.add "correct_answer:", I18n.t("messages.validate_answer_size_correct")
+      end
+      correct_answer_size = self.word_answers.size
+      if Settings.answer_default > correct_answer_size ||
+        correct_answer_size > Settings.max_answer
+        errors.add "answer_size:", I18n.t("messages.validate_answer_size")
+      end
+    end
 
+    def verify_used_word
```
Comment (dieunb) [inline app/models/word.rb]: indent

File: app/views/admin/words/_form.html.erb [hunk 5]
```diff
@@ -1,11 +1,19 @@
 <%= form_for [:admin, @word] do |f| %>
   <%= render "/shared/error_messages", object: f.object %>
+  <div class="col-md-2 col-md-offset-1">
+    <% if @categories.present? %>
+      <label><%= t "category.categories" %></label>
+        <%= f.select "category_id",
```
Comment (dieunb) [inline app/views/admin/words/_form.html.erb]: indent

**Pull request merged**